dataservice/api/participant/resources.py

```diff
@@ -1,114 +1,74 @@
-from flask import request
-from flask_restplus import Namespace, Resource
+from flask import abort, request
+from flask.views import MethodView
 
 from dataservice.extensions import db
-from dataservice.api.participant import models
-from dataservice.api.common.formatters import kf_response
+from dataservice.api.participant.models import Participant
+from dataservice.api.participant.schemas import ParticipantSchema
 
-description = open('dataservice/api/participant/README.md').read()
 
-participant_api = Namespace(name='participants', description=description)
-
-from dataservice.api.participant.serializers import (  # noqa
-    participant_fields,
-    participant_list,
-    participant_response
-)
-
-
-@participant_api.route('/')
-class ParticipantList(Resource):
-    @participant_api.marshal_with(participant_list)
-    def get(self):
+class ParticipantAPI(MethodView):
+    def get(self, kf_id):
         """
-        Get all participants
+        Get a participant by id
 
-        Returns paginated participants
+        Gets a participant given a Kids First id
         """
-        participants = models.Participant.query.all()
-        return kf_response(participants, 200, '{} participants'.
-                           format(len(participants)))
+        if kf_id is None:
+            return (ParticipantSchema(many=True)
+                    .jsonify(Participant.query.all()))
+        else:
+            try:
+                participant = Participant.query.filter_by(kf_id=kf_id).one()
+            except:
+                abort(404, 'could not find {} `{}`'
+                      .format('Participant', kf_id))
+            return ParticipantSchema().jsonify(participant)
 
-    @participant_api.marshal_with(participant_response)
-    @participant_api.doc(responses={201: 'participant created',
-                                    400: 'invalid data'})
-    @participant_api.expect(participant_fields)
     def post(self):
         """
         Create a new participant
-
-        Creates a new participant and assigns a Kids First id
         """
         body = request.json
-        participant = models.Participant(**body)
-        db.session.add(participant)
+        p = Participant(external_id=body.get('external_id'))
```

Comment:
Isn't this where you would use the marshmallow schema to load from dict to your model object?

Reply:
Yup, getting to it :)